opendebates/admin.py

```diff
@@ -15,9 +17,33 @@ class CategoryAdmin(ModelAdmin):
 @register(models.Submission)
 class SubmissionAdmin(ModelAdmin):
     list_display = [f.name for f in models.Submission._meta.fields]
-    actions = [admin_list_export]
+    list_filter = ('approved', )
+    search_fields = ('idea', )
+    actions = [admin_list_export, 'remove_submissions']
     raw_id_fields = ['voter', 'duplicate_of']
 
+    def remove_submissions(self, request, queryset):
+        "Custom action to mark submissions 'unapproved' and to notify users by email."
+        if request.POST.get('post'):
+            count = 0
+            # only email user if submission changes status, hence the filter
+            for submission in queryset.filter(approved=True):
+                count += 1
+                submission.approved = False
+                submission.moderated_removal = True
+                submission.removal_flags.all().update(reviewed=True)
+                submission.save()
```

Comment:
Yes, I actually wrote a version with just those queries (and the first one needs to get a list of PKs so we can email later), but then reverted it to this because I thought this was clearer to read :) I didn't think a couple hundred queries was a problem since it's an admin action and not likely to happen frequently, but definitely see your point. Jeff, any preference? (he's taken over this from me)